ui/shared/components/form/ReduxFormWrapper.jsx

```diff
@@ -268,7 +271,11 @@ class ReduxFormWrapper extends React.Component {
   }
 
   componentWillUpdate(nextProps) {
-    if (nextProps.submitSucceeded && nextProps.closeOnSuccess && !nextProps.noModal) {
+    if (this.props.onSubmitSucceeded && nextProps.submitSucceeded && nextProps.closeOnSuccess && nextProps.noModal) {
```

Comment:
I think if you pass an `onSubmitSucceeded` callback and submit succeeded you should call the callback but not call `handleClose` at all. So I don't think you need to check `closeOnSuccess` or `noModal` here